src/main/java/mServer/crawler/sender/ard/ArdVideoDeserializer.java

```diff
@@ -21,74 +23,122 @@
     private static final String JSON_ELEMENT_WIDTH = "_width";
     private static final String JSON_ELEMENT_HEIGHT = "_height";
     
+    private static String addMissingHttpPrefixIfNecessary(String aUrl) {
+      if(aUrl.startsWith("//")) {
+        return TEXT_START_HTTP + ":" + aUrl;
+      }
+      
+      return aUrl;
+    }
+    
     @Override
     public ArdVideoDTO deserialize(JsonElement aJsonElement, Type aType, JsonDeserializationContext aContext) throws JsonParseException {
         ArdVideoDTO dto = new ArdVideoDTO();
         
         if (aJsonElement.isJsonObject()) {
             JsonObject jsonObject = aJsonElement.getAsJsonObject();
             if (jsonObject.has(JSON_OBJECT_MEDIAARRAY)) {
-                Map<Qualities, UrlWidthHeightDTO> urlMap = new HashMap<>();
+                Map<Qualities, String> urlMap = new HashMap<>();
                 
                 JsonArray mediaArray = jsonObject.get(JSON_OBJECT_MEDIAARRAY).getAsJsonArray();
-                mediaArray.forEach(mediaArrayItem -> {
-                    JsonObject item = mediaArrayItem.getAsJsonObject();
-                    JsonArray streamArray = item.get(JSON_OBJECT_MEDIASTREAMARRAY).getAsJsonArray();
-                    streamArray.forEach(streamItem -> {
-                        deserializeMediaStreamArrayItem(streamItem.getAsJsonObject(), urlMap);
-                    });
-                });
+                if (mediaArray.size() > 0) {
+                  JsonElement mediaArrayItem = mediaArray.get(mediaArray.size() - 1);
+                  JsonObject item = mediaArrayItem.getAsJsonObject();
+                  JsonArray streamArray = item.get(JSON_OBJECT_MEDIASTREAMARRAY).getAsJsonArray();
+                  deserializeMediaStreamArray(streamArray, urlMap);
+                }
                 
                 // URLs setzen
                 urlMap.forEach((key, value) -> {
-                    dto.addVideo(key, value.url);
+                    dto.addVideo(key, value);
                 });
             }
         }
         
         return dto;
     }    
     
-    private void deserializeMediaStreamArrayItem(JsonObject stream, Map<Qualities, UrlWidthHeightDTO> urlMap) {
+    private void deserializeMediaStreamArray(JsonArray streamArray, Map<Qualities, String> urlMap) {
+      List<QualityUrlWidthHeightDTO> qualities = new ArrayList<>();
+      
+      streamArray.forEach(streamItem -> {
+        qualities.add(deserializeUrlInfos(streamItem.getAsJsonObject()));
+      });
+      
+      convertQualities(qualities, urlMap);
+    }
+    
+    private void convertQualities(List<QualityUrlWidthHeightDTO> qualities, Map<Qualities, String> urlMap) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)